backend/LexCore/Entities/Project.cs

```diff
@@ -8,7 +8,7 @@ namespace LexCore.Entities;
 public class Project : EntityBase
 {
 
-    public const string ProjectCodeRegex = @"^[a-z-\d]+$";
+    public const string ProjectCodeRegex = @"^[a-z\d][a-z-\d]*$";
```

Comment:
since we're changing this can we write a couple unit tests in `HgServiceTests`?

Reply:
Done, and I realized that HgService.InitRepo wasn't actually checking against the project code regex, so I added a check there as well.